dimod/constrained.py

```diff
@@ -652,6 +626,8 @@ def set_objective(self, objective: Union[BinaryQuadraticModel, QuadraticModel]):
             >>> cqm.set_objective(2*i - 0.5*i*j + 10)
 
         """
+        if isinstance(objective, Iterable):
```

Comment:
I think this works, but `not isinstance(objective, [QuadraticModel, BinaryQuadraticModel)` is likely to be a tiny bit more future-proof...